common/app/redux/index.js

```diff
@@ -238,6 +238,9 @@ export const isSignedInSelector = state => !!userSelector(state).username;
 export const challengeSelector = state => {
   const challengeName = currentChallengeSelector(state);
   const challengeMap = entitiesSelector(state).challenge || {};
+  if (!challengeMap[challengeName]) {
+    return {};
+  }
   return challengeMap[challengeName] || firstChallengeSelector(state);
```

Comment:
is firstChallengeSelector ever going to get called now?

Reply:
Accessing freeCodeCamp's challenges still take me to the first "learn how freecodecamp works" challenge for me, meaning firstChallengeSelector was called.

Reply:
Yes because in that case challengeMap[challengeName] will not be 'undefined'. actually the above code in simplicity is
```
return challengeMap[challengeName] || {};
```
According to my understanding firstChallengeSelector will never get called. If there is requirement for it @Bouncey  can clear it.